src/Collection/Resources/assets/src/controller.ts

```diff
@@ -0,0 +1,98 @@
+import { Controller } from '@hotwired/stimulus';
+
+const DEFAULT_ITEMS_SELECTOR = ':scope > :is(div, fieldset)';
+
+interface CollectionDataset extends DOMStringMap {
+    prototype: string;
+    currentIndex: string;
+    itemsSelector?: string;
+    addButtonTemplateId?: string;
+    disableAddButton?: string;
+    deleteButtonTemplateId?: string;
+    disableDeleteButton?: string;
+}
+
+enum ButtonType {
+    Add = 'add',
+    Delete = 'delete',
+}
+
+export default class extends Controller {
+    connect() {
+        this.connectCollection(this.element as HTMLElement);
+    }
+
+    connectCollection(parent: HTMLElement) {
+        parent.querySelectorAll('[data-prototype]').forEach((el) => {
+            const collectionEl = el as HTMLElement;
+            const items = this.getItems(collectionEl);
+            collectionEl.dataset.currentIndex = items.length.toString();
+
+            this.addAddButton(collectionEl);
+
+            this.getItems(collectionEl).forEach((itemEl) => this.addDeleteButton(collectionEl, itemEl as HTMLElement));
+        });
+    }
+
+    getItems(collectionElement: HTMLElement) {
+        return collectionElement.querySelectorAll(collectionElement.dataset.itemsSelector || DEFAULT_ITEMS_SELECTOR);
+    }
+
+    createButton(collectionEl: HTMLElement, buttonType: ButtonType): HTMLElement {
+        const buttonTemplateID = collectionEl.dataset[`${buttonType}ButtonTemplateId`];
+        if (buttonTemplateID && 'content' in document.createElement('template')) {
+            // Get from template
+            const buttonTemplate = document.getElementById(buttonTemplateID) as HTMLTemplateElement | null;
+            if (!buttonTemplate) throw new Error(`element with ID "${buttonTemplateID}" not found`);
+
+            return buttonTemplate.content.cloneNode(true) as HTMLElement;
+        }
+
+        // If no template is provided, create a raw HTML button
+        const button = document.createElement('button') as HTMLButtonElement;
+        button.type = 'button';
+        button.textContent = buttonType === ButtonType.Add ? 'Add' : 'Delete';
+
+        return button;
+    }
+
+    addItem(collectionEl: HTMLElement) {
+        const currentIndex = (collectionEl.dataset as CollectionDataset).currentIndex;
+        (collectionEl.dataset.currentIndex as unknown as number)++;
+
+        const collectionNamePattern = collectionEl.id.replace(/_/g, '(?:_|\\[|]\\[)');
+
+        const prototype = (collectionEl.dataset.prototype as string) // We're sure that dataset.prototype exists, because of the CSS selector used in connect()
+            .replace('__name__label__', currentIndex)
```

Comment:
Actually my code supports nested collections even when not using another placeholder, but I'll add support for this option!

Reply:
No it does not. The prototype of the outer collection actually contains the attribute storing the prototype of the inner collection (as it contains the whole inner collection). If you use the same placeholder for both, the replacement done here will replace both placeholders inside the inner prototype, breaking the inner collection.

Reply:
replacing only the first occurrence is also broken. `__name__` will appear multiple time in the prototype when the entry type is a compound type (as the name of each child form will have the placeholder in its name.

Reply:
The first occurence in every string: https://github.com/symfony/ux/pull/398/files#diff-0c84303846c77e8d3377b23960aed4a1dfc3db0bb6206c7435ef6fe890a7fa1bR49

Unless I'm missing something, this works with the current example I provided for test purpose.

Reply:
@dunglas AFAICT, this would replace both placecholders in `name="root[outer_collection][__name__][inner_collection][__name][field_1]"`
And trying to do clever tricks based on the quote delimiting attributes (to replace the first occurrence in each attribute) is a bad idea: in the prototype attribute of the nested collection, those would be HTML entities and not actual quotes.

Note that you need to add multiple items in each collection to notice the issue (if you only add one outer item with a single inner item, you won't notice whether the `[0]` come from the replacements done by the outer or inner item)